spec/models/infra_conversion_job_spec.rb

```diff
@@ -70,18 +343,88 @@
       it_behaves_like 'allows error signal'
 
       it_behaves_like 'doesn\'t allow start signal'
+      it_behaves_like 'doesn\'t allow collapse_snapshots signal'
     end
   end
 
-  context 'operations' do
-    let(:poll_interval) { Settings.transformation.job.retry_interval }
-
+  context 'transition methods' do
     context '#start' do
       it 'to poll_automate_state_machine when preflight_check passes' do
         expect(job).to receive(:queue_signal).with(:poll_automate_state_machine)
         job.signal(:start)
         expect(task.reload.state).to eq('migrate')
-        expect(task.options[:workflow_runner]).to eq('automate')
+      end
+    end
+
+    context '#collapse_snapshots' do
+      let(:async_task) { FactoryBot.create(:miq_task, :userid => user.id) }
+      let(:snapshot) { FactoryBot.create(:snapshot, :vm_or_template => vm_vmware) }
```

Comment:
Just to make things a little more interesting, I would change this to:

`let(:snapshots) { FactoryBot.create_list(:snapshot, 2, :vm_or_template => vm_vmware) }